Sources/FluentKit/Schema/SchemaBuilder.swift

```diff
@@ -38,6 +38,39 @@ public final class SchemaBuilder {
         ))
         return self
     }
+
+    public func foreignKey(_ field: String, references parent: (table: String, field: String), onUpdate: DatabaseSchema.Constraint.ForeignKeyAction = .noAction, onDelete: DatabaseSchema.Constraint.ForeignKeyAction = .noAction) -> Self {
+        self.schema.constraints.append(.foreignKey(
+            fields: [.string(field)],
+            parentTable: parent.table,
+            parentFields: [.string(parent.field)],
+            updateAction: onUpdate,
+            deleteAction: onDelete
+        ))
+        return self
+    }
+
+    public func foreignKey(_ fields: (String, String), references parent: (table: String, fields: (String, String)), onUpdate: DatabaseSchema.Constraint.ForeignKeyAction = .noAction, onDelete: DatabaseSchema.Constraint.ForeignKeyAction = .noAction) -> Self {
```

Comment:
What's the benefit of creating a single foreign key that references multiple fields over just creating two foreign keys?

Reply:
I've not used composite foreign keys before, so forgive the possibility that I misrepresent it here. My understanding is that, because a foreign key must reference a column that is primary or unique in the parent table, a composite primary key could generally only be referenced by a composite foreign key. Additionally, two separate foreign keys (I think) would mean the value for key 1 matches _some_ column in the parent and the value for key 2 matches _some_ column in the parent, but not necessarily in the same row, whereas a composite key of two columns means both column values match the same row in the parent.

Reply:
Unless I'm mistaken, it seems like Laravel at least doesn't support this: https://stackoverflow.com/questions/39736866/composite-foreign-key-laravel

Can you point to any other ORMs that support it?

Reply:
I'm entirely unfamiliar with Laravel, and their documentation definitely does not describe it, but their does appear to be support at least at a low level for any number of columns in a foreign key table constraint (via an alter statement): https://github.com/laravel/framework/blob/1bbe5528568555d597582fdbec73e31f8a818dbc/src/Illuminate/Database/Schema/Grammars/Grammar.php#L65

At any rate, this seems like the kind of feature that is not used often but if you want it then it's a bummer that your ORM of choice does not support it. That being said, I am certainly willing to rip it out. If I rip out the support for composite foreign keys, I do not think there is much benefit in supporting foreign key *table* constraints at all. We could just move forward with foreign key *column* constraints to reduce the breadth of code changes here (for example, the changes to `SQLTableConstraint` were only made to facilitate foreign key *table* constraints). We might need to support table constraints that `SQLTableConstraint` cannot represent in the future _anyway_, but that time could potentially not be now.

Reply:
I did a little more looking and was able to find examples (like [Doctrine](https://www.doctrine-project.org/projects/doctrine-orm/en/2.6/tutorials/composite-primary-keys.html)) of composite foreign key support in ORMs, but I think ultimately I might actually land on the "don't support" side of this debate simply because fluent does not support composite primary keys (at least not right now) so there seems to be incredibly minimal utility in being able to define composite foreign keys.

Reply:
> I think ultimately I might actually land on the "don't support" side of this debate simply because fluent does not support composite primary keys

Yeah that's what I'm thinking, too. This seems like it should be added as a todo item to the composite keys issue. If / when Fluent does get composite keys, there might end up being a clearer way to support this in migrations. 

Until then, I think it still makes sense for the underlying `DatabaseSchema` types to support an array of fields even if the helper APIs don't use that. That's one less thing to worry about for when we do add composite keys and it means users can write their own extensions for adding composite key fields if they want.